src/picosvg/svg.py

```diff
@@ -381,111 +479,154 @@ def _combine_clip_paths(self, clip_paths: Sequence[SVGPath]) -> SVGPath:
+                child_context = SVGTraverseContext(
+                    nth_of_type,
+                    child,
+                    path,
+                    transform,
+                    clips,
+                    _attrib_to_pass_on(child, context.attrib),
                 )
-                new_el.attrib["id"] = self._new_id("clipPath", "merged-clip-%d")
-                new_el.append(to_element(clip_path))
-                new_clip_paths[clip_path.d] = new_el
+                frontier.append(child_context)
 
-            new_ref_id = new_clip_paths[clip_path.d].attrib["id"]
-            clipped_el.attrib["clip-path"] = f"url(#{new_ref_id})"
+    def _transformed_gradient(self, url, transform):
+        fill_el = self.resolve_url(url, "*")
+        assert _is_gradient(fill_el), f"Not sure how to fill from {fill_el.tag}"
 
-        # destroy unreferenced clip paths
-        for old_clip_path in old_clip_paths:
-            if old_clip_path.getparent() is None:
-                continue
-            old_id = old_clip_path.attrib["id"]
-            if not self.xpath(f'//svg:*[@clip-path="url(#{old_id})"]'):
-                _safe_remove(old_clip_path)
+        new_fill = copy.deepcopy(fill_el)
+        new_fill.attrib["gradientTransform"] = (
+            _element_transform(fill_el, transform)
+            .round(_GRADIENT_TRANSFORM_NDIGITS)
+            .tostring()
+        )
+        new_fill_id = self._new_id(fill_el.attrib["id"] + "_%d")
+        new_fill.attrib["id"] = new_fill_id
 
-    def _compute_clip_path(self, el):
-        """Resolve clip path for element, including inherited clipping.
+        fill_el.addnext(new_fill)
+        return new_fill
 
-        None if there is no clipping.
+    def _simplify(self):
+        """
+        Removes groups where possible, applies transforms, applies clip paths.
 
-        https://www.w3.org/TR/SVG11/masking.html#EstablishingANewClippingPath
+        Groups with 0 < opacity < 1 and multiple children are retained.
         """
-        clip_paths = []
-        while el is not None:
-            clip_url = el.attrib.get("clip-path", None)
-            if clip_url:
-                clip_paths.append(self._resolve_clip_path(clip_url))
-            el = el.getparent()
+        # Reversed: we want leaves first
+        to_process = reversed(tuple(c for c in self.breadth_first()))
 
-        if not clip_paths:
-            return None
-        return self._combine_clip_paths(clip_paths)
+        for context in to_process:
+            # print(context.path, len(context.clips), "clips")
+            if "clipPath" in context.path:
+                _safe_remove(context.element)
+                continue
+
+            el = context.element
+            _del_attrs(el, "clip-path", "transform")  # handled separately
+
+            skips = {"clip-path", "transform", "opacity"}  # handled separately
+            _inherit_attrib(context.attrib, el, skips=skips)
```

Comment:
I see that `transform` attribute is skipped here as it's handled separately; however the function `_attrib_to_pass_on` still collects them, multiply them, and pass them down and stores them into context.attrib; maybe `_attrib_to_pass_on` should use the same list of `skips` so it doesn't do unnecessary work more than once?

Reply:
Stripped clip-path and transform out as having special handling in traversal.